### PR TITLE
fix(docx-io): handle broken tokens due to whitespace in docx export

### PR DESCRIPTION
**Pull request by arthrod**
The DOCX export process was failing to parse tracking tokens (especially comment start tokens) when they were broken by line wrapping or whitespace insertion in the HTML output. This caused tokens to appear as text in the document and comment metadata (like author name) to be missing.

This commit updates `DOCX_TOKEN_REGEX` to robustly handle whitespace within token payloads (`[\s\S]+?`) and modifies `parseDocxToken` to strip whitespace from the payload before decoding.

Also added a regression test in `packages/docx-io/src/lib/html-to-docx/tracking.spec.ts` that simulates a token payload broken by whitespace inside a URI escape sequence.

---
*PR created automatically by Jules for task [12862530258887572815](https://jules.google.com/task/12862530258887572815) started by @arthrod*

## Summary by Sourcery

Improve robustness of DOCX tracking token parsing when tokens are split or contain whitespace.

Bug Fixes:
- Fix DOCX tracking tokens failing to parse when payloads are broken by whitespace or line wrapping in the exported HTML.

Enhancements:
- Refine tracking-related type definitions and internal tracking structures for comments and suggestions to better represent payloads and stored comments.

Tests:
- Add a regression test ensuring DOCX comment tokens are correctly parsed when their URI-encoded payload is interrupted by whitespace.

### CODE REVIEW

**Comment by arthrod** [comment]
> **Actionable comments posted: 1**
> 
> 🤖 Fix all issues with AI agents
> ```
> In `@packages/docx-io/src/lib/html-to-docx/tracking.spec.ts`:
> - Around line 428-447: The new test "should handle token payload broken by
> whitespace" is located outside any describe block; move this it(...) block into
> the existing describe('Round-trip Token Encoding', ...) scope (or create a new
> describe for whitespace handling) so it is properly grouped; locate the it(...)
> with its payload construction and assertions and cut/paste it inside the
> describe that contains other round-trip encoding tests so
> splitDocxTrackingTokens is tested within that describe.
> ```

@jules, evaluate the comments carefully. Assess with skepticism. Please review and implement carefully, step by step. Ensure to test and document your changes.

File: packages/docx-io/src/lib/html-to-docx/tracking.ts [hunk 1]
```diff
@@ -160,29 +147,31 @@ function parseDocxToken(
...
-      return { type: 'insEnd', id: decoded };
+      return { id: decoded, type: 'insEnd' };
     }
 
     const data = JSON.parse(decoded);
 
     if (kind === 'CMT') {
-      return { type: 'commentStart', data: data as CommentPayload };
+      return { data: data as CommentPayload, type: 'commentStart' };
     }
```
[the remark this answers is not in the document]
Comment (arthrod) [reply packages/docx-io/src/lib/html-to-docx/tracking.ts]: > security-medium medium
> 
> The `parseDocxToken` function parses a URI-encoded JSON payload from a tracking token without validating its structure. If an attacker provides a payload where the `replies` field is not an array but has a `length` property (e.g., a string or an object like `{"replies": {"length": 1}}`), it will cause a `TypeError` when the payload is processed in `buildRunsFromTextWithTokens` (in `xml-builder.ts`), leading to a denial of service by crashing the document export process.
> 
> Specifically, in `xml-builder.ts`, the code checks `data.replies && data.replies.length > 0` and then calls `data.replies.forEach(...)`. If `data.replies` is a string, `length > 0` is true, but `forEach` is undefined, causing a crash. You should validate that `replies` is an array if present.
> 
> ```ts
>     const data = JSON.parse(decoded);
> 
>     if (kind === 'CMT') {
>       if (data.replies && !Array.isArray(data.replies)) return null;
>       return { data: data as CommentPayload, type: 'commentStart' };
>     }
> ```

@jules, evaluate the comments carefully. Assess with skepticism. Please review and implement carefully, step by step. Ensure to test and document your changes.
